src/ptvsd/adapter/messages.py

```diff
@@ -256,15 +272,18 @@ def _configure(self, request):
             # before it had a chance to send the event - so wake up periodically, and
             # check whether server channel is still alive.
             while not debuggee.wait_for_pid(1):
-                if _channels.server() is None:
+                if not self._no_debug and _channels.server() is None:
```

Comment:
There's one interesting new possibility here: if the launcher crashes or gets killed before it can communicate the PID back to us, this loop will never exit.

This isn't entirely new. Previously, if the server crashed while we were waiting for "process", that would also close the socket, and we'd detect it with the channel going away. But if it crashed before it connected, then we'd sit forever in `accept_connection_from_server()`.

So it looks like we need to add some timeouts. Since you're already in this code, can you limit the number of times this loop will spin, before it bails out and fails the request? Or actually maybe just convert it to a simple wait with a timeout... it's not really expected for the server to crash before it sends "process", so if we take a while longer to detect that, it's okay, so long as we don't hang.

And then `accept_connection_from_server()` just needs to call `listener.settimeout(...)` before `accept()`.

Reply:
What is a good timeout? say about 10 seconds.

Reply:
It's all arbitrary, but I think if it can't start and connect within 10 seconds, the debug session is going to be unusably slow anyway :)